client/landing/stepper/declarative-flow/flows/ai-site-builder/ai-site-builder.ts

```diff
@@ -36,16 +36,28 @@ const deletePage = async ( siteId: string | number, pageId: number ): Promise< b
 };
 
 function initialize() {
-	// stepsWithRequiredLogin will take care of redirecting to the login step if the user is not logged in.
-	return stepsWithRequiredLogin( [
-		STEPS.SITE_CREATION_STEP,
-		STEPS.PROCESSING,
-		STEPS.ERROR,
-		shouldRenderRewrittenDomainSearch() ? STEPS.DOMAIN_SEARCH : STEPS.UNIFIED_DOMAINS,
-		STEPS.UNIFIED_PLANS,
-		STEPS.SITE_LAUNCH,
-		STEPS.PROCESSING,
-	] as const );
+	// If spec_id is present, skip the SiteSpec widget and go directly to site creation
+	// If not present, show the SiteSpec widget to collect specifications
+	const queryParams = new URLSearchParams( window.location.search );
+	const hasSpecId = queryParams.get( 'spec_id' );
+
+	if ( hasSpecId ) {
+		// Skip SiteSpec widget, go directly to site creation
+		return [
+			...stepsWithRequiredLogin( [
+				STEPS.SITE_CREATION_STEP,
+				STEPS.PROCESSING,
+				STEPS.ERROR,
+				shouldRenderRewrittenDomainSearch() ? STEPS.DOMAIN_SEARCH : STEPS.UNIFIED_DOMAINS,
+				STEPS.UNIFIED_PLANS,
+				STEPS.SITE_LAUNCH,
+				STEPS.PROCESSING,
+			] ),
+		];
+	}
+
+	// Show SiteSpec widget to collect specifications
+	return [ STEPS.SITE_SPEC ];
```

Comment:
Shouldn't we add the other steps like the site-creation, processing etc?

Reply:
You're right! Addressed in [7ee4c12](https://github.com/Automattic/wp-calypso/pull/105543/commits/7ee4c12ab03465507b8bb46869b0398d2d2d4651)